[1.20.1] Log spam upon player death

#### Issue Description:
With the Blood Magic & Curios API mods installed, when the player dies, an exception is thrown and printed to the log file repeatedly.

#### What happens:
The  following stack trace is printed to the log file repeatedly:

```
[13:08:21] [Render thread/ERROR] [ne.mi.cl.gu.ov.ForgeGui/]: Error rendering overlay 'minecraft:chat_panel'
java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:143) ~[?:?] {re:mixin}
	at wayoftime.bloodmagic.compat.CuriosCompat.getCuriosInventory(CuriosCompat.java:26) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at wayoftime.bloodmagic.compat.CuriosCompat.lambda$registerInventory$0(CuriosCompat.java:20) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at wayoftime.bloodmagic.util.helper.InventoryHelper.lambda$getActiveInventories$2(InventoryHelper.java:75) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at java.util.HashMap.forEach(HashMap.java:1421) ~[?:?] {re:mixin}
	at wayoftime.bloodmagic.util.helper.InventoryHelper.getActiveInventories(InventoryHelper.java:75) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at wayoftime.bloodmagic.util.Utils.canPlayerSeeDemonWill(Utils.java:576) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at wayoftime.bloodmagic.client.hud.element.ElementDemonAura.shouldRender(ElementDemonAura.java:78) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at wayoftime.bloodmagic.client.hud.ElementRegistry.onRenderOverlay(ElementRegistry.java:140) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:rei_plugin_compatibilities:B}
	at wayoftime.bloodmagic.client.hud.__ElementRegistry_onRenderOverlay_Pre.invoke(.dynamic) ~[bloodmagic-1.20.1-3.3.3-45.jar%23225!/:1.20.1-3.3.3-45] {re:classloading,pl:eventbus:B}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73) ~[eventbus-6.0.5.jar%23133!/:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%23133!/:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%23133!/:?] {}
	at net.minecraftforge.client.gui.overlay.ForgeGui.pre(ForgeGui.java:643) ~[forge-1.20.1-47.1.105-universal.jar%23289!/:?] {re:classloading,pl:rei_plugin_compatibilities:B}
	at net.minecraftforge.client.gui.overlay.ForgeGui.lambda$render$0(ForgeGui.java:125) ~[forge-1.20.1-47.1.105-universal.jar%23289!/:?] {re:classloading,pl:rei_plugin_compatibilities:B}
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422) ~[guava-31.1-jre.jar%2394!/:?] {re:mixin}
	at net.minecraftforge.client.gui.overlay.ForgeGui.m_280421_(ForgeGui.java:121) ~[forge-1.20.1-47.1.105-universal.jar%23289!/:?] {re:classloading,pl:rei_plugin_compatibilities:B}
	at net.minecraft.client.renderer.GameRenderer.m_109093_(GameRenderer.java:949) ~[client-1.20.1-20230612.114412-srg.jar%23284!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1146) ~[client-1.20.1-20230612.114412-srg.jar%23284!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftClientMixin,pl:mixin:APP:architectury.mixins.json:MixinMinecraft,pl:mixin:APP:flywheel.mixins.json:PausedPartialTickAccessor,pl:mixin:APP:carryon.mixins.json:MinecraftMixin,pl:mixin:APP:notenoughcrashes.mixins.json:client.MixinMinecraftClient,pl:mixin:APP:ars_nouveau.mixins.json:light.ClientMixin,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:botania_xplat.mixins.json:client.MinecraftAccessor,pl:mixin:APP:create.mixins.json:client.WindowResizeMixin,pl:mixin:APP:ars_nouveau.mixins.json:camera.MinecraftMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:718) ~[client-1.20.1-20230612.114412-srg.jar%23284!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftClientMixin,pl:mixin:APP:architectury.mixins.json:MixinMinecraft,pl:mixin:APP:flywheel.mixins.json:PausedPartialTickAccessor,pl:mixin:APP:carryon.mixins.json:MinecraftMixin,pl:mixin:APP:notenoughcrashes.mixins.json:client.MixinMinecraftClient,pl:mixin:APP:ars_nouveau.mixins.json:light.ClientMixin,pl:mixin:APP:iceberg.mixins.json:MinecraftMixin,pl:mixin:APP:botania_xplat.mixins.json:client.MinecraftAccessor,pl:mixin:APP:create.mixins.json:client.WindowResizeMixin,pl:mixin:APP:ars_nouveau.mixins.json:camera.MinecraftMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:218) ~[minecraft-1.20.1-client.jar:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:rei_plugin_compatibilities:B,pl:mixin:APP:flywheel.mixins.json:ClientMainMixin,pl:mixin:APP:notenoughcrashes.forge.mixins.json:client.MixinMain,pl:mixin:A,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:126) ~[loader-47.2.2.jar:47.2] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:114) ~[loader-47.2.2.jar:47.2] {}
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.runService(CommonClientLaunchHandler.java:24) ~[loader-47.2.2.jar:47.2] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:108) ~[loader-47.2.2.jar:47.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:67) ~[?:?] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100) ~[?:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) ~[?:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[?:?] {}
```

#### What you expected to happen:
The exception is prevented from being thrown

#### Steps to reproduce:

1. Create a new instance in prism launcher
2. Target minecraft 1.20.1 and neoforge 47.1.1.03
3. Download the BloodMagic and Curios API mods from modrinth
4. Start the game
5. in Prism Launcher, Click edit and then minecraft log, to view the log file
6. Create a new world with cheats enabled
7. run the `/kill` command.
8. Check the log file in Prism

____
#### Affected Versions (Do *not* use "latest"):

- BloodMagic: 3.3.3-45
- Curios API: 5.9.0+1.20.1
- Patchouli: 1.20.1-84-FORGE
- Minecraft: 1.20.1
- NeoForge: 47.1.103
- LWJGL: 3.3.1
- Prism Launcher 8.3

Same problem while death in 3.3.3-45 forge today.

Neruina output(partly):
net.minecraft.ReportedException: Ticking player
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerPlayer.m_9240_(ServerPlayer.java:530)
...
Caused by: java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:143)
	at TRANSFORMER/bloodmagic@3.3.3-45/wayoftime.bloodmagic.compat.CuriosCompat.getCuriosInventory(CuriosCompat.java:26)
...

Hope for a new release to solve this problem.

--- a/src/main/java/wayoftime/bloodmagic/compat/CuriosCompat.java
+++ b/src/main/java/wayoftime/bloodmagic/compat/CuriosCompat.java
@@ -23,7 +23,11 @@ public class CuriosCompat
 
 	public NonNullList<ItemStack> getCuriosInventory(Player player)
 	{
-		IItemHandlerModifiable itemHandler = CuriosApi.getCuriosInventory(player).resolve().get().getEquippedCurios();
+		Optional<> curioInv = CuriosApi.getCuriosInventory(player).resolve();
+		if (!curioInv.isPresent()) {
+			return NonNullList.create();
+		}
+		IItemHandlerModifiable itemHandler = curioInv.get().getEquippedCurios();
 		NonNullList<ItemStack> inventory = NonNullList.create();
 		for (int i = 0; i < itemHandler.getSlots(); i++)
 		{

--- a/src/main/java/wayoftime/bloodmagic/compat/CuriosCompat.java
+++ b/src/main/java/wayoftime/bloodmagic/compat/CuriosCompat.java
@@ -6,12 +6,14 @@ import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraftforge.items.IItemHandlerModifiable;
 import top.theillusivec4.curios.api.CuriosApi;
+import top.theillusivec4.curios.api.type.capability.ICuriosItemHandler;
 import top.theillusivec4.curios.api.type.inventory.ICurioStacksHandler;
 import wayoftime.bloodmagic.core.LivingArmorRegistrar;
 import wayoftime.bloodmagic.core.living.LivingStats;
 import wayoftime.bloodmagic.core.living.LivingUtil;
 import wayoftime.bloodmagic.impl.BloodMagicAPI;
 import java.util.Map;
+import java.util.Optional;
 
 public class CuriosCompat
 {
@@ -23,8 +25,8 @@ public class CuriosCompat
 
 	public NonNullList<ItemStack> getCuriosInventory(Player player)
 	{
-		Optional<> curioInv = CuriosApi.getCuriosInventory(player).resolve();
-		if (!curioInv.isPresent()) {
+		Optional<ICuriosItemHandler> curioInv = CuriosApi.getCuriosInventory(player).resolve();
+		if (curioInv.isEmpty()) {
 			return NonNullList.create();
 		}
 		IItemHandlerModifiable itemHandler = curioInv.get().getEquippedCurios();

